snapcraft/internal/sources/errors.py

```diff
@@ -21,6 +21,26 @@ class VCSError(errors.SnapcraftError):
     fmt = '{message}'
 
 
+class SnapcraftSourceUnhandledError(errors.SnapcraftError):
+
+    fmt = ('Failed to pick an appropriate source handler:\n'
+           '{source} does not look like a known format.\n'
+           'Check that the URL is correct and consider adding "source-type".')
+
+    def __init__(self, source):
+        super().__init__(source=source)
+
+
+class SnapcraftSourceNotADirectoryError(errors.SnapcraftError):
+
+    fmt = ('Failed to pick an appropriate source handler:\n'
```

Comment:
this message is more confusing than helpful

Reply:
Maybe:
```
Failed to get source: {source!} is not a directory.
Run `snapcraft help sources` to read the documentation
```

Reply:
Making both say the same generic suggestion seems less helpful to me... though I can totally be convinced if you guys like it. Would go for a single error class `SnapcraftSourceUnhandledError` for both cases then.

Reply:
Correction. The second half of the first part is different as I've just come to realize... I feel like I really need a line break in there.

Reply:
The style suggested by mpt was:
```
generic error: specific reason \n
how to fix
```
We still need to make the other errors consistent with this. And because in here so many things could have gone wrong, like a typo, missing source-type, a type that we don't support yet, etc. I think the appropriate action to fix this would be for the user to read the docs and figure it out by themselves. But as I said, just a suggestion. 

And the error messages is something we'll have to iterate over and over again, making small improvements. During the sprint we talked to mpt about how to do it, so I think it's good to just give it our best and then get corrections from him now and then.

Reply:
I like Leo's suggestion here, although I'd personally change `get` to `pull`.

Reply:
I went with a modified one based on the input on the other error, including "pulling the source for part {part_name!r}".